From 22fa6b8789b8ecb204402d47cea907d2803d694c Mon Sep 17 00:00:00 2001
From: Ariel Aviad <aaviad@habana.ai>
Date: Wed, 14 May 2025 11:04:25 +0300
Subject: [PATCH] [SW-223975] u-boot: fix reset after reading flash

We didn't wait after sending wfe to device and
in some cases called reset in recovery mode before
the cpu finished to enter wfe and write the boot_info
to SRAM this caused the boot info to be corrupted.

This adds wait for wfe that to prevent potential data corruption.

Change-Id: Id966ed934cb9d1461afcc10a5a981dc87ffd59ce
Signed-off-by: Ariel Aviad <aaviad@habana.ai>
---

diff --git a/src/uboot.cpp b/src/uboot.cpp
index a97a707..4b10d1a 100644
--- a/src/uboot.cpp
+++ b/src/uboot.cpp
@@ -656,6 +656,8 @@
 	uboot_lkd_wfc(UBOOT_CLR_TIMEOUT);
 
 	uboot_send_lkd_cmd(COMMS_NOOP, 0);
+
+	pci_wait_for_wfe();
 }
 
 int uboot_dev_send_reset_dev(enum proto_type proto)
